Formula/helix.rb

```diff
@@ -0,0 +1,22 @@
+class Helix < Formula
+  desc "Post-modern modal text editor"
+  homepage "https://helix-editor.com"
+  url "https://github.com/helix-editor/helix.git",
+       tag:      "v0.6.0",
+       revision: "ac1b7d8d0a608f47edfee2872d414e94fd26cc31"
+  license "MPL-2.0"
+  head "https://github.com/helix-editor/helix.git", branch: "master"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "helix-term")
+    prefix.install "runtime"
+    mv bin/"hx", bin/"hx-bin"
+    (bin/"hx").write_env_script(bin/"hx-bin", HELIX_RUNTIME: prefix/"runtime")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/hx --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The case for Helix is a little bit different. Actually the output of `hx --help` is 
```
helix-term v0.6.0
Blaž Hrastnik <blaz@mxxn.io>
A post-modern text editor.

USAGE:
    hx [FLAGS] [files]...

ARGS:
    <files>...    Sets the input file to use

FLAGS:
    -h, --help       Prints help information
    --tutor          Loads the tutorial
    -v               Increases logging verbosity each use for up to 3 times
                     (default file: /Users/henry/.cache/helix/helix.log)
    -V, --version    Prints version information
```

Unlike `vim` or `nevim`, Helix has not yet provide a headless mode, which enables the test script to perform operations without blocking the execution. I think `-h` or `-v` are currently the only two options that can output something without block the execution.

Alternatively, do you think it a good idea if we run `hx -v` and check whether the log file exists?

Reply:
If it's a program that runs in the terminal can't you just open it, wait a bit and send an exit command?

Reply:
Thanks for the suggestions. Looks like Helix does not provide a way for sending commands externally. I am not sure whether `SIGINT` is considered as one. I may need some references, is there any other formulas having a test script doing that?

Reply:
See for example https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/kondo.rb

Reply:
Thanks but unfortunately, writing to `stdin` can not tell Helix to exit since it handles `stdin` its own way. Currently the way I do the testing is that:
- On macOS, I try to write some content to `stdin` and check for the error message produced.
- On Linux, I start the program, write something to `stdin` (the content will be in the editor buffer), wait for 5 seconds, make sure the process is still alive (otherwise `kill` will exit with 1), and kill it.